### PR TITLE
feat: Added missing analytics events and  extra data

### LINKED ISSUE #15367: [Task]: Missing backend analytics events
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

With the audit logs project, these missing backend events for some core actions have to be added

Workspace
* Creating a workspace
* Deleting a workspace

App
* Importing an app
* Exporting an app
* Cloning an app

Page
* Viewing a page

User
* Log in
* Log out

### PR DESCRIPTION
**Pull request by vishnu-gp**
## Description

This adds a few missing analytics events and some more extra data for audit purposes. The missing event added are,

- User login (login_USER)
- User logout (logout_USER)
- Import Application (import_APPLICATION)
- Export Application (export_APPLICATION)
- Clone Application (clone_APPLICATION)
- Create Workspace (create_WORKSPACE)
- Delete Workspace (delete_WORKSPACE)
- View Page (view_NEWPAGE)

Fixes #15367 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manually on local

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by mohanarpit — APPROVED** [review]

**Review by trishaanand — APPROVED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java [hunk 1]
```diff
@@ -1138,4 +1140,26 @@ private Mono<Application> sendCloneApplicationAnalyticsEvent(Application sourceA
...
+     * To send analytics event for page views
+     * @param newPage Page being accessed
+     * @param viewMode Page is accessed in view mode or not
+     * @return NewPage
+     */
+    private Mono<NewPage> sendPageViewAnalyticsEvent(NewPage newPage, boolean viewMode) {
+        if (!viewMode) {
+           return Mono.empty();
+        }
+
```
Comment (vishnu-gp) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java:1153]: @AnaghHegde I was trying to get the application from here.

**Comment by vuiets** [comment]
@vishnu-gp could we change the names of a few of the events to remove redundant words
login_USER
logout_USER
import_APPLICATION
export_APPLICATION
clone_APPLICATION

**Review by trishaanand — DISMISSED** [review]

**Comment by vishnu-gp** [comment]
> 



> @vishnu-gp could we change the names of a few of the events to remove redundant words login_USER logout_USER import_APPLICATION export_APPLICATION clone_APPLICATION

@vuiets import_application_APPLICATION and export_application_APPLICATION already existed in the system even before this PR, Do you see any complications if I change that to this new suggested format?

**Comment by vuiets** [comment]
Good question. We'll have to track the data separately for imports and exports starting with the this change since Mixpanel doesn't allow us to merge labels. 

Don't see any data getting collected for `export_application_APPLICATION` on Mixpanel so renaming it shouldn't be a problem.

For import, I see another event called `IMPORT_API`, can you confirm if this is a BE event or FE event?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java [hunk 2]
```diff
@@ -1107,5 +1114,52 @@ public Mono<Application> createOrUpdateSuffixedApplication(Application applicati
...
+    /**
+     * To send analytics event for page views
+     * @param newPage Page being accessed
+     * @param viewMode Page is accessed in view mode or not
+     * @return NewPage
+     */
+    private Mono<NewPage> sendPageViewAnalyticsEvent(NewPage newPage, boolean viewMode) {
+        if (!viewMode) {
+           return Mono.empty();
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java:1151]: Minor nit: Formatting seems to be off here. Can you please check once?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java [hunk 3]
```diff
@@ -781,7 +785,10 @@ public Mono<Application> cloneApplication(String applicationId, String branchNam
                                                         application.getId(), editModeThemeId, publishedModeThemeId, MANAGE_APPLICATIONS
                                                 ).thenReturn(application);
                                             })
-                            );
+                            )
+                            .flatMap(application -> {
+                                return sendCloneApplicationAnalyticsEvent(sourceApplication, application);
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java:790]: Can we remove the return statement here and use the lambda?

**Review by AnaghHegde — COMMENTED** [review]
Minor changes. Rest looks good to me.

**Comment by vishnu-gp** [comment]
> IMPORT_API

@vuiets Couldn't find any BE event named IMPORT_API, probably an FE one. I will make the changes here.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java (hunk 2 above)
Comment (vishnu-gp) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java]: @AnaghHegde Corrected

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java (hunk 3 above)
Comment (vishnu-gp) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java]: @AnaghHegde Done

**Review by AnaghHegde — APPROVED** [review]

**Comment by vuiets** [comment]
Thanks then, please rename `export_application_APPLICATION` to `export_APPLICATION` as well.

**Comment by vishnu-gp** [comment]
> Thanks then, please rename `export_application_APPLICATION` to `export_APPLICATION` as well.

@vuiets That's already done and updated in the PR description.

**Pull request merged**